docs/Constraints.rst

```diff
@@ -0,0 +1,45 @@
+Constraints Class
+=================
+
+It manages the constraints of the problem. It is used to store the constraints and to check if a solution satisfies them.
+
+Constraints types supported
+---------------------------
+
+Types of constraints supported:
+
+- *Equality*, writing the equality as a quadratic penalty function, i.e. sum = b imposed with g = (sum-b)^2
+- *Inequality*, moving to equality with the continuous auxiliary variables a to be expanded with the encoding technique of the Variables class
+- *Not constraint* among binary variables, i.e. Not(a) = b imposed with g = 2ab - a - b + 1
+- *And constraint* among binary variables, i.e.  a and b = c imposed with  ab -2(a+b)c + 3c
+- *Or constraint* among binary variables, i.e. a or b = c imposed with ab + (a+b)(1-2c) + c
+- *Xor constraint* among binary variables, i.e. a xor b = c imposed with 2ab - 2(a+b)c - 4(a+b)\_aux+4_aux c +a+b+c+4+\_aux
+
+Constraints declarations
+----------------------
```

Comment:
Please make sure that proper RST headlines are used.
The underlines must be as long as the header in all cases.
Sphinx should tell you that and scream at you when you try to build the documentation (`nox -s docs`)

When I try to build the docs, the following errors and warnings pop up:
```bash
/home/lburgholzer/PycharmProjects/mqt-qao/docs/Constraints.rst:19: WARNING: Title underline too short.

Constraints declarations
----------------------
/home/lburgholzer/PycharmProjects/mqt-qao/docs/Constraints.rst:19: WARNING: Title underline too short.

Constraints declarations
----------------------
/home/lburgholzer/PycharmProjects/mqt-qao/docs/Problem.rst:20: ERROR: Unexpected indentation.
/home/lburgholzer/PycharmProjects/mqt-qao/docs/Problem.rst:28: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/lburgholzer/PycharmProjects/mqt-qao/docs/Solution.rst:42: WARNING: Bullet list ends without a blank line; unexpected unindent.
/home/lburgholzer/PycharmProjects/mqt-qao/docs/Solver.rst:38: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/lburgholzer/PycharmProjects/mqt-qao/docs/Solver.rst:51: ERROR: Unexpected indentation.
/home/lburgholzer/PycharmProjects/mqt-qao/docs/Solver.rst:76: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/lburgholzer/PycharmProjects/mqt-qao/docs/Solver.rst:94: ERROR: Unexpected indentation.
/home/lburgholzer/PycharmProjects/mqt-qao/docs/Solver.rst:111: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/lburgholzer/PycharmProjects/mqt-qao/docs/Solver.rst:121: ERROR: Unexpected indentation.
/home/lburgholzer/PycharmProjects/mqt-qao/docs/Solver.rst:141: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/lburgholzer/PycharmProjects/mqt-qao/docs/Solver.rst:154: ERROR: Unexpected indentation.
/home/lburgholzer/PycharmProjects/mqt-qao/docs/Solver.rst:173: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/lburgholzer/PycharmProjects/mqt-qao/docs/Solver.rst:184: ERROR: Unexpected indentation.
/home/lburgholzer/PycharmProjects/mqt-qao/docs/index.rst:26: WARNING: toctree contains reference to nonexisting document 'Variables'
/home/lburgholzer/PycharmProjects/mqt-qao/docs/index.rst:26: WARNING: toctree contains reference to nonexisting document 'Solvers'
looking for now-outdated files... none found
pickling environment... done
checking consistency... /home/lburgholzer/PycharmProjects/mqt-qao/docs/Solver.rst: WARNING: document isn't included in any toctree
/home/lburgholzer/PycharmProjects/mqt-qao/docs/Variable.rst: WARNING: document isn't included in any toctree
```

Please fix those.

Reply:
Done